### PR TITLE
feat: add rover controller to landing page

### PR DESCRIPTION
**Pull request by olivierwilkinson**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready | [OS-1219](https://pi-top.atlassian.net/browse/OS-1219) |

#### move landing page template to components

Move LandingPageTemplate out of pages folder into components. Since the
template is used to create pages and is not a page itself it should live
in components.

#### add rover controller to landing page

Create a systemd service for the rover controller. Using systemd gives
us an API for starting and stopping the server while keeping the
controller decoupled from the web portal.

Provide routes on the backend for enabling, disabling and getting the
status of the systemd service. The systemd service status can be
'active' while the controller server is still starting up so verify the
server is ready before responding with an active status.

Add RoverControllerLanding component which can launch and stop the
controller. Show loading states when launching and stopping the
controller and provide troublshooting when the controller crashes.

Modify the tsconfig to allow implicit returns since they are common in
effects that conditionally need to cleanup as can be seen in the
RoverControllerLanding component.

### CODE REVIEW

File: frontend/src/components/roverControllerLanding/__tests__/RoverControllerLanding.test.tsx [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import { act } from "react-dom/test-utils";
+
+const matchers = {
+  initialising: "Getting rover controller status...",
+  stopped: textContentMatcher(
+    /Have adventures with your very own pi-top \[4\] Mars rover\!Control your rover directly from your computer or phone/
+  ),
+  started: /Rover controller is running\!/,
+  launching: /Launching rover controller/,
+  running: /Rover controller is running/,
```
Comment (olivierwilkinson) [inline frontend/src/components/roverControllerLanding/__tests__/RoverControllerLanding.test.tsx:22]: Note to self: remove dupe matcher

**Comment by jcapona** [comment]
nice, works great for me! I do think we can improve how the 'open rover controller' link is displayed... maybe a button would be better? 

image

also! if you run this in the landing app that's displayed on first boot, clicking the 'open rover controller' link doesn't open anything... I think you can use [this helper](https://github.com/pi-top/pi-topOS-Web-Portal/blob/cee2ea8fde7033f603f79ac997536168411a1269/frontend/src/helpers/utils.ts#L2 ) to determine if you're running on the device, then use a service like [this](https://github.com/pi-top/pi-topOS-Web-Portal/blob/master/frontend/src/services/openKnowledgeBase.ts#L5) that would make the backend [open a new window with a particular URL](https://github.com/pi-top/pi-topOS-Web-Portal/blob/master/pt_os_web_portal/backend/helpers/landing.py#L34 ) in the actual desktop of the Raspberry Pi.

Also, you can run the "first boot" landing app in the device running `pt-os-web-portal-frontend landing` in a terminal.

**Comment by olivierwilkinson** [comment]
> nice, works great for me! I do think we can improve how the 'open rover controller' link is displayed... maybe a button would be better?
> 
> image
> 
> also! if you run this in the landing app that's displayed on first boot, clicking the 'open rover controller' link doesn't open anything... I think you can use [this helper](https://github.com/pi-top/pi-topOS-Web-Portal/blob/cee2ea8fde7033f603f79ac997536168411a1269/frontend/src/helpers/utils.ts#L2) to determine if you're running on the device, then use a service like [this](https://github.com/pi-top/pi-topOS-Web-Portal/blob/master/frontend/src/services/openKnowledgeBase.ts#L5) that would make the backend [open a new window with a particular URL](https://github.com/pi-top/pi-topOS-Web-Portal/blob/master/pt_os_web_portal/backend/helpers/landing.py#L34) in the actual desktop of the Raspberry Pi.
> 
> Also, you can run the "first boot" landing app in the device running `pt-os-web-portal-frontend landing` in a terminal.

Yeah the link looks a bit naff doesn't it? I've updated it to use the styling we have for links in challenges :) let me know what you think!

For the web renderer most of the time the user will be using a screen and keyboard so showing the rover controller doesn't make as much sense. It's also hidden after onboarding is finished so they are unlikely to have a built rover at that point. Instead I've put a warning message saying that they should go to pi-top.local on a different device if that works for you?

**Review by jcapona — APPROVED** [review]
Yeah, I agree that users shouldn't really run this in their pi-top so the new message makes sense. Also, the 'open rover controller' button is way better now!

**Pull request merged**